documentation/src/test/java/org/hibernate/search/documentation/search/predicate/PredicateDslIT.java

```diff
@@ -872,6 +873,128 @@ void simpleQueryString() {
 					.extracting( Book::getId )
 					.containsExactlyInAnyOrder( BOOK1_ID, BOOK3_ID );
 		} );
+
+		withinSearchSession( searchSession -> {
+			// tag::simpleQueryString-minimum-should-match[]
+			List<Book> hits = searchSession.search( Book.class )
+					.where( f -> f.queryString().field( "title" )
+							.matching( "crime robot investigate automatic detective" )
+							.minimumShouldMatchNumber( 2 ) )
+					.fetchHits( 20 );
+			// end::simpleQueryString-minimum-should-match[]
+			assertThat( hits )
+					.extracting( Book::getId )
+					.containsExactlyInAnyOrder( BOOK4_ID );
+		} );
+	}
+
+	@Test
+	void queryString() {
+		withinSearchSession( searchSession -> {
+			// tag::queryString-query[]
+			List<Book> hits = searchSession.search( Book.class )
+					.where( f -> f.queryString().field( "description" )
+							.matching(
+									"robots +(crime || investigation || disappearance)^10 +\"investigation help\"~2 -/(dis)?a[p]+ea?ance/" ) ) // <1>
```

Comment:
hmm right, well I guess in this particular case it wouldn't matter anyway, since it'll get converted to should clauses so I'll just remove that 😃 .... but I was looking at https://lucene.apache.org/core/9_9_2/queryparser/org/apache/lucene/queryparser/classic/package-summary.html#OR (and https://lucene.apache.org/core/9_9_2/queryparser/org/apache/lucene/queryparser/classic/package-summary.html#Escaping_Special_Characters)
and also here https://www.elastic.co/guide/en/elasticsearch/reference/8.12/query-dsl-query-string-query.html#_boolean_operators 
and I thought just to throw lots of different things into that query (even if some may not make much sense) just to show that the query can be a bit harder to read 😃

Reply:
Hu, that's interesting. So `||` *is* supported. Please ignore me, then.